### ISSUE TITLE
Missing MIME type

### ISSUE THREAD

**Issue by DavidMStraub** [gramps-project/gramps-webapi#168]
On my system, when trying to generate a Graphviz report (with Graphviz installed), I get

```
  File "gramps_webapi/api/resources/reports.py", line 212, in get
    return send_file(buffer, mimetype=types_map[file_type])
KeyError: '.gv'
```

If I try to set the mimetype to `application/octet-stream`, I get

```
File "lib/python3.8/site-packages/flask/helpers.py", line 592, in send_file
    raise ValueError(
ValueError: Unable to infer MIME-type because no filename is available. Please set either `attachment_filename`, pass a filepath to `filename_or_fp` or set your own MIME-type via `mimetype`.
```

Not sure what's the best way to deal with this; possibly setting `appliction/octet-stream` *plus* setting an explicit filename in case the MIME type is not recognized.

**Comment by cdhorn** [comment]
A couple things at play here...

First, most of this is probably a lack of knowledge on my part on proper mime handling and I made a few mistakes.  I don't know `Graphviz` and in testing I ran into:
```
>>> types_map[".dot"]
'application/msword'
```
That clearly is not a `Graphviz` file and some searches all kept turning up Microsoft Word.  So at the time I foolishly assumed that  `.gv` and `.dot` were interchangeable so remapped the one type to the other which I now realize they are not. So I need to fix that.

I also had trouble with `.gvpdf` and `.gspdf` as well as they were not in `types_map` on my system at all so I remapped them to `.pdf` and I now wonder if that was incorrect to do as well.

Now, oddly, I can't seem to reproduce your error on my system using the API but I can from a normal Python shell which does not seem to make sense. The `/usr/share/mime/text/vnd.graphviz.xml` is certainly there and I don't see why it would be recognized in one context and not the other.  And having found that I see the `.dot` mapping to `application/msword` must take precedence over the other file.

It may make sense to switch back to keeping our own dictionary of types Gramps outputs for lookups like I was originally doing. I know you said that was redundant with `types_map` and it seems it should be hence I changed it, but that would at least present consistent behaviour.  Any thoughts on why I'm seeing inconsistent behaviour?

**Comment by DavidMStraub** [comment]
Yeah it's confusing, I don't understand it fully either. I saw there is [mimetypes.init()](https://docs.python.org/3/library/mimetypes.html#mimetypes.init) which might add some additional types (so calling this or not might lead to different behaviour), but I don't know if this has anything to do with the inconsistencies you're seeing.

**Comment by DavidMStraub** [comment]
In principle, I'm not opposed to hardcode mime types.

**Comment by cdhorn** [comment]
> Yeah it's confusing, I don't understand it fully either. I saw there is mimetypes.init() which might add some additional types (so calling this or not might lead to different behaviour), but I don't know if this has anything to do with the inconsistencies you're seeing.

Well that resolves the `.gv` lookup issue in a Python shell and interestingly `.dot` maps properly too.  But now remember why I thought they are one and the same, the generated report files should end in `.dot` but instead Gramps or whatever it calls to create them is appending a `.gv` extension: `007b1999-66d0-437b-85b2-3b6587214d18.dot.gv`

**Comment by cdhorn** [comment]
Seems to imply they are the same:
[https://www.graphviz.org/doc/info/output.html#d:gv](https://www.graphviz.org/doc/info/output.html#d:gv)

**Comment by DavidMStraub** [comment]
Solved by #171.
